### PR TITLE
feat: use new srv to capture screen shot and movie

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

I added new srv type to capture screen shot or movie by external trigger.

```
uint8 SCREEN_SHOT = 0
uint8 START_BUFFERING = 1
uint8 STOP_BUFFERING = 2
uint8 RECORD = 3
uint8 SAVE = 4
uint8 SAVE_BUFFER = 5

uint8 action
string file_name

---

bool success
```

### screen shot

```
ros2 service call /debug/capture tier4_screen_capture_rviz_plugin/srv/Capture "{action: 0, file_name: "test"}"
```

https://github.com/user-attachments/assets/95848bc1-d340-4ae6-88a8-84e1be767ad8

### capture movie

start recording
```
ros2 service call /debug/capture tier4_screen_capture_rviz_plugin/srv/Capture "{action: 3, file_name: ""}"
```
save
```
ros2 service call /debug/capture tier4_screen_capture_rviz_plugin/srv/Capture "{action: 4, file_name: "test-2"}"
```

https://github.com/user-attachments/assets/5b5ddda0-d37d-4cb6-ad37-0bd2e625d311

### buffering screen capture

start buffering
```
ros2 service call /debug/capture tier4_screen_capture_rviz_plugin/srv/Capture "{action: 1, file_name: ""}"
```
save
```
ros2 service call /debug/capture tier4_screen_capture_rviz_plugin/srv/Capture "{action: 5, file_name: "buffer"}"
```

https://github.com/user-attachments/assets/b44c283c-addd-492f-8b7d-35afd6e4e226

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by kyoichi-sugahara — APPROVED** [review]
The following are the results of the operation check:
- No issues with operation check from the rviz GUI
- No issues with operation check by sending service from the terminal
- After executing START_BUFFERING, no data was saved when SAVE was executed
- After executing RECORD, no data was saved when SAVE_BUFFER was executed
Copy

LGTM! Thank you so much for correcting bad implementation fot "buffer" feature

**Review by Naophis — APPROVED** [review]
LGTM!!

File: control_data_collecting_tool/scripts/data_collecting_pure_pursuit_trajectory_follower.py [hunk 1]
```diff
@@ -526,9 +526,9 @@ def control(self):
 
         # [2] publish cmd
         control_cmd_msg = AckermannControlCommand()
-        control_cmd_msg.stamp = (
-            control_cmd_msg.lateral.stamp
-        ) = control_cmd_msg.longitudinal.stamp = (self.get_clock().now().to_msg())
```
Comment (kosuke55) [inline control_data_collecting_tool/scripts/data_collecting_pure_pursuit_trajectory_follower.py:531]: (seems just formatting)

**Review by kosuke55 — APPROVED** [review]

**Comment by satoshi-ota** [comment]
```
      Start 26: cpplint_rosidl_generated_py

26: Test command: /usr/bin/python3 "-u" "/opt/ros/humble/share/ament_cmake_test/cmake/run_test.py" "/__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/test_results/tier4_screen_capture_rviz_plugin/cpplint_rosidl_generated_py.xunit.xml" "--package-name" "tier4_screen_capture_rviz_plugin" "--output-file" "/__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/ament_cpplint/cpplint_rosidl_generated_py.txt" "--command" "/opt/ros/humble/bin/ament_cpplint" "--xunit-file" "/__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/test_results/tier4_screen_capture_rviz_plugin/cpplint_rosidl_generated_py.xunit.xml" "--filters=-readability/fn_size" "--linelength" "999" "--root" "/__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py" "/__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin"
26: Test timeout computed to be: 120
26: -- run_test.py: invoking following command in '/__w/autoware_tools/autoware_tools/common/tier4_screen_capture_rviz_plugin':
26:  - /opt/ros/humble/bin/ament_cpplint --xunit-file /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/test_results/tier4_screen_capture_rviz_plugin/cpplint_rosidl_generated_py.xunit.xml --filters=-readability/fn_size --linelength 999 --root /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_c.c:36:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_c.c:38:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_c.c:43:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_c.c:45:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_c.c:178:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_c.c:180:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_c.c:185:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_c.c:187:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_c.c:298:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_c.c:306:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_fastrtps_c.c:36:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_fastrtps_c.c:38:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_fastrtps_c.c:43:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_fastrtps_c.c:45:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_fastrtps_c.c:178:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_fastrtps_c.c:180:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_fastrtps_c.c:185:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_fastrtps_c.c:187:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_fastrtps_c.c:298:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_fastrtps_c.c:306:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_introspection_c.c:36:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_introspection_c.c:38:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_introspection_c.c:43:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_introspection_c.c:45:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_introspection_c.c:178:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_introspection_c.c:180:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_introspection_c.c:185:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_introspection_c.c:187:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_introspection_c.c:298:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_introspection_c.c:306:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/srv/_capture_s.c:24:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/srv/_capture_s.c:53:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/srv/_capture_s.c:85:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/srv/_capture_s.c:90:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/srv/_capture_s.c:101:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/srv/_capture_s.c:151:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/srv/_capture_s.c:180:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/srv/_capture_s.c:197:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/srv/_capture_s.c:202:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/srv/_capture_s.c:213:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
26: Category 'whitespace/line_length' errors found: 40
26: Total errors found: 40
26: Using '--root=/__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py' argument
26: 
26: Done processing /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_c.c
26: 
26: Done processing /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_fastrtps_c.c
26: 
26: Done processing /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/_tier4_screen_capture_rviz_plugin_s.ep.rosidl_typesupport_introspection_c.c
26: 
26: Done processing /__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/rosidl_generator_py/tier4_screen_capture_rviz_plugin/srv/_capture_s.c
26: 
26: -- run_test.py: return code 1
26: -- run_test.py: verify result file '/__w/autoware_tools/autoware_tools/build/tier4_screen_capture_rviz_plugin/test_results/tier4_screen_capture_rviz_plugin/cpplint_rosidl_generated_py.xunit.xml'
26/35 Test #26: cpplint_rosidl_generated_py .......................***Failed    0.36 sec
```

... :thinking:

**Pull request merged**